src/lib/protocols/iso9506-1-mms.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * iso9506-1-mms.c
+ *
+ * ISO 9506-1:2003 Manufacturing Message Specification
+ * 
+ * Copyright (C) 2023 - ntop.org
+ * Copyright (C) 2023 - V.G <jacendi@protonmail.com>
+ *
+ * This file is part of nDPI, an open source deep packet inspection
+ * library based on the OpenDPI and PACE technology by ipoque GmbH
+ *
+ * nDPI is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * nDPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with nDPI.  If not, see <http://www.gnu.org/licenses/>.
+ * 
+ */
+
+#include "ndpi_protocol_ids.h"
+
+#define NDPI_CURRENT_PROTO NDPI_PROTOCOL_ISO9506_1_MMS
+
+#include "ndpi_api.h"
+#include "ndpi_private.h"
+
+static void ndpi_int_iso9506_1_mms_add_connection(struct ndpi_detection_module_struct *ndpi_struct,
+                                                struct ndpi_flow_struct *flow)
+{
+  NDPI_LOG_INFO(ndpi_struct, "found ISO 9506-1 MMS\n");
+  ndpi_set_detected_protocol(ndpi_struct, flow,
+                             NDPI_PROTOCOL_ISO9506_1_MMS, NDPI_PROTOCOL_UNKNOWN,
+                             NDPI_CONFIDENCE_DPI);
+}
+
+static void ndpi_search_iso9506_1_mms(struct ndpi_detection_module_struct *ndpi_struct,
+                                    struct ndpi_flow_struct *flow)
+{
+  struct ndpi_packet_struct const * const packet = &ndpi_struct->packet;
+
+  NDPI_LOG_DBG(ndpi_struct, "search ISO 9506-1 MMS\n");
+  
+  if ((packet->payload_packet_len > 60) && (packet->payload[0] == 3) &&
+      (packet->payload[1] == 0) && 
+      (get_u_int16_t(packet->payload, 2) == htons(packet->payload_packet_len)))
+  {
+    if (current_pkt_from_client_to_server(ndpi_struct, flow)) {
+      /* Check COTP and ISO 8327-1 headers */
```

Comment:
Did you try the `subprotocol` feature?
AFAIS, it may not work with your code. But LGTM anyway.